theseus/extlib/baspacho_solver.h

```diff
@@ -0,0 +1,68 @@
+#pragma once
+
+#include <torch/extension.h>
+
+#include "baspacho/baspacho/Solver.h"
+
+struct SymbolicDecompositionData {
+    BaSpaCho::SolverPtr solver;
+
+#ifdef THESEUS_HAVE_CUDA
+    bool isCuda;
+#endif
+    torch::Tensor toParamIndex;
+    torch::Tensor paramStart;
+    torch::Tensor paramSize;
+};
+
+// Numeric decomposition, CPU version
+class NumericDecomposition {
```

Comment:
Maybe helpful to leave a brief comment next to all class/functions about what they do.

Reply:
+1.

Reply:
done!